python/pyarrow/_parquet.pyx

```diff
@@ -592,6 +596,16 @@ cdef class ParquetWriter:
         else:
             props.disable_deprecated_int96_timestamps()
 
+    cdef void _set_coerce_timestamps(self,
+                                     ArrowWriterProperties.Builder* props):
+        if self.coerce_timestamps == 'ms':
+            props.coerce_timestamps(TimeUnit_MILLI)
+        elif self.coerce_timestamps == 'us':
+            props.coerce_timestamps(TimeUnit_MICRO)
+        elif self.coerce_timestamps is not None:
+            raise ValueError('Invalid value for coerce_timestamps: {0}'
```

Comment:
good catch. if you have int return value and `except -1` then I believe cython handles automatically returning -1 when you raise an exception